csv-generator/tests/consumers_test/manuscript_consumer_test.py

```diff
@@ -32,11 +32,29 @@ def test_will_handle_no_msid(manuscript_consumer: ManuscriptXMLConsumer):
 def test_can_process_data(mock_write_row: MagicMock,
                           manuscript_consumer: ManuscriptXMLConsumer,
                           manuscript_xml: str):
-    expected = ['1526868166', 'test_file.zip', 'foobar.xml',
+    expected = ['1526868166', 'foobar.xml',
                 '33099', 'United States', '10.7554/eLife.33099']
 
     root = etree.fromstring(manuscript_xml)
     manuscript_consumer.process(root, 'foobar.xml')
 
     assert mock_write_row.called_once()
     assert mock_write_row.call_args == call(expected)
+
+
+def test_source_file_names(manuscript_xml: str):
+    source_file_name = 'bar.xml'
+    cases = [
+        (None, source_file_name), # nothing at all
+        ('foo.zip', 'foo.zip/%s' % source_file_name), # zip
+        ('foo.csv', 'foo.csv'), # csv
+        ('foo.wtf', source_file_name), # unknown ext
+        ('foo', source_file_name), # no ext
+    ]
+    create_date = 'doesntmatter'
+    for given_zip_file_name, expected_fname in cases:
+        with better_temp_dir() as tempdir:
+            klass = ManuscriptXMLConsumer(create_date, given_zip_file_name, tempdir)
```

Comment:
minor: `klass` doesn't refer to a class but an instance and probably should be renamed to `instance`?